docs/ldml/tr35-keyboards.md

```diff
@@ -2412,6 +2416,287 @@ Used in the `to=`
 
     Emits the named mark. Also see [Markers](#markers).
 
+#### Transform Grammar
+
+##### Transform From Grammar
+
+The `from=` attribute MUST match the `from-match` rule in this grammar. Not all strings which match this grammar are valid:
+
+* It is an error if there are more than 9 capture groups
+* Unicode codepoints and escaped characters may not be invalid or unpaired surrogates
+* The CLDR repository may define additional requirements on the repertoire, such as requiring all characters to be in a published Unicode version and disallowing private-use characters.
+* It is an error to reference variables that are not defined.
+
+The following is the [LDML EBNF](./tr35.md#ebnf) format for the grammar:
+
+```ebnf
+
+from-match
+         ::= '^'? atoms
+atoms    ::= atom ( '|'? atom )*
+atom     ::= quark quantifier?
+quark    ::= non-group
+           | group
+non-group
+         ::= simple-matcher
+           | codepointseq
+           | variable
+variable ::= string-variable
+           | set-variable
+string-variable
+         ::= '${' var-id '}'
+set-variable
+         ::= '$[' var-id ']'
+group    ::= capturing-group
```

Comment:
Is there a supposed ordering to the grammar? It feels like the idea is to use first, define next, on depth-first basis, but then why is this not followed by `var-id` and then `codepointseq`? I don't mind whether its define-then-use or use-then-define or alphabetical, but it would help if it was systematic.

Reply:
There isn't a specific order. I can put them in some better order.